### ISSUE TITLE
[BUG] Installing a package from GitHub appears to send NPM into an indefinite, high CPU usage frenzy

### ISSUE THREAD

**Issue by brianjenkins94** [npm/cli#3787]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

High CPU usage with no sign of stopping.

### Expected Behavior

The package gets installed.

### Steps To Reproduce

```
npm install brettz9/rollup-plugin-postprocess#update --save-dev
```

### Environment

- OS: macOS 10.15.7 (19H524)
- Node: v16.9.1
- npm: 7.23.0

**Comment by lukekarrys** [comment]
Thanks for the issue! I've confirmed this happens on the latest npm `v7.24.0` and I was able to get an error log by `SIGINT`-ing the process at the first sign it was hitting an infinite loop: [2021-09-23T17_46_08_912Z-debug.log](https://github.com/npm/cli/files/7219820/2021-09-23T17_46_08_912Z-debug.log).

This seems to point to an issue within arborist as it loops on `placeDep` when preparing the git dep. This might be a duplicate of https://github.com/npm/cli/issues/3385 but I'm keeping this open since this one install of a git dep always fails for me.
